### PR TITLE
Some clippy fixes for 1.36

### PR DESCRIPTION
**Pull request by kjeremy**
Some clippy fixes now that 1.36 is released. ~~Plus the requisite format run (I can rebase after #1481 is merged to make this cleaner) .~~

The change from `map(|it| *it)` to `copied()` changes the minimum rust stable to 1.36.

### CODE REVIEW

File: crates/ra_hir/src/lang_item.rs [hunk 1]
```diff
@@ -142,7 +142,7 @@ impl LangItems {
     {
         let node = item.source(db).ast;
         if let Some(lang_item_name) = lang_item_name(&*node) {
-            self.items.entry(lang_item_name).or_insert(constructor(item));
+            self.items.entry(lang_item_name).or_insert_with(|| constructor(item));
```
Comment (kjeremy) [inline crates/ra_hir/src/lang_item.rs]: Looks like this might prevent some accidental work.

**Comment by kjeremy** [comment]
Rebased for formatting.

**Comment by matklad** [comment]
Awesome, thanks!

bors r+

**Pull request merged**